### PR TITLE
Fix the warnings complained by Clang

### PR DESCRIPTION
**Pull request by ufownl**
This PR fixes the warning complained by Clang and the file name conflicts on case-insensitive systems.

### CODE REVIEW

File: util/threading.h [hunk 1]
```diff
@@ -225,23 +222,23 @@ class PerClusterPools {
...
+    return outer_pool_->NumWorkers() * inner_pools_[0]->NumWorkers();
   }
 
  private:
   bool have_threading_support_;
   CoreBitSets cores_per_cluster_;
-  hwy::ThreadPool outer_pool_;
   // hwy::ThreadPool is unfortunately not marked as movable, so we have to use
   // unique_ptr.
+  std::unique_ptr<hwy::ThreadPool> outer_pool_;
```
[the remark this answers is not in the document]
Comment (ufownl) [reply util/threading.h:241]: This is to maintain the move semantics of `PerClusterPools`. If that is unnecessary, we can delete the move constructor and operator of `PerClusterPools`, as Clang recommends.

[the remark this answers is not in the document]
Comment (jan-wassenberg) [reply util/threading.h:241]: I see, thanks :) Yes, deleting sounds like a good alternative. This class is more or less a singleton and I don't expect it to be moved.

File: util/threading.h [hunk 2]
```diff
@@ -114,15 +114,11 @@ class PerClusterPools {
...
-  // caller of `ThreadPool::Run`, which is the outer worker or main thread.
+  // Returns the number of threads for `ThreadPool`: zero if there is no
+  // threading support, otherwise the capped number of workers.
   size_t CappedNumThreads(size_t num_workers, size_t user_max_or_zero) const {
     if (!have_threading_support_) return 0;
-    const size_t capped_num_workers =
-        CapIfNonZero(num_workers, user_max_or_zero);
-    // Avoid underflow if number of workers is zero.
-    return capped_num_workers == 0 ? 0 : capped_num_workers - 1;
+    return CapIfNonZero(num_workers, user_max_or_zero);
```
[the remark this answers is not in the document]
Comment (ufownl) [reply util/threading.h:121]: When creating the inner pools, the number of threads in the inner pools is calculated by `CappedNumThreads(cores_per_cluster_[outer].Count(), max_innter_workers)`. If `max_inner_workers` is `0`, should the number of threads in the inner pools be `cores_per_cluster_[outer].Count() - 1`? My understanding is that the number of threads in the inner pool should be `cores_per_cluster_[outer].Count()`, including the thread of the outer pool calling `Run` of the inner pool.

File: util/threading.h [hunk 3]
```diff
@@ -135,7 +131,7 @@ class PerClusterPools {
     const size_t max_threads_per_outer = max_threads / outer_workers;
     // First outer pool gets the remainder.
     const size_t remainder = (outer == 0) ? (max_threads % outer_workers) : 0;
-    return 1 + max_threads_per_outer + remainder;
+    return HWY_MAX(max_threads_per_outer + remainder, 1);
```
[the remark this answers is not in the document]
Comment (ufownl) [reply util/threading.h:134]: Sorry, I missed the documentation about `max_threads`, if it's semantics like that my version is indeed wrong. So the root of this issue is what I mentioned above, right?

[the remark this answers is not in the document]
Comment (jan-wassenberg) [reply util/threading.h:134]: Yes indeed :) Sorry about the confusing semantics, it is unfortunately too late to change it in ThreadPool :(

File: util/threading.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (jan-wassenberg) [reply util/threading.h:121]: > If max_inner_workers is 0, should the number of threads in the inner pools be cores_per_cluster_[outer].Count() - 1

Yes, that's right.

The difference is that the thread calling pool.Run() also counts as a worker, so the ThreadPool only creates N-1 *extra* threads alongside that first worker, which makes for a total of N workers.

[the remark this answers is not in the document]
Comment (ufownl) [reply util/threading.h:121]: I realize my earlier expression may have been a bit unclear. What I actually want to confirm is whether the number of workers in the inner pool should be `cores_per_cluster_[outer].Count()`. In the current implementation, the number of workers is `cores_per_cluster_[outer].Count() - 1`, which seems counterintuitive to me.

[the remark this answers is not in the document]
Comment (jan-wassenberg) [reply util/threading.h:121]: :) Let me clarify. We mean different things by worker and thread. Imagine a cluster as a CCX with 8 HW cores. One of those should run one of the OS threads belonging to the outer pool. The other 7 cores run 7 extra threads spawned by the inner pool. Thus there are 8 (== `cores_per_cluster_[outer].Count()`) *workers* in the inner pool, though it has only spawned 7 *threads*. Num workers == num threads + 1.
Does that make sense?

[the remark this answers is not in the document]
Comment (ufownl) [reply util/threading.h:121]: I checked the latest implementation of `hwy::ThreadPool`, `num_workers_` changed from `HWY_MAX(num_threads, size_t{1})` to `num_threads + 1`. If we use the latest version of Highway, this issue will not exist. :)

**Review by jan-wassenberg — APPROVED** [review]

**Pull request merged**